examples/xdp_drop_count.rs

```diff
@@ -0,0 +1,110 @@
+extern crate bcc;
+
+use bcc::{
+    BPFBuilder, BccError,
+    XDPMode::{XDP_FLAGS_HW_MODE, XDP_FLAGS_SKB_MODE},
+    XDP,
+};
+use byteorder::{LittleEndian, ReadBytesExt};
+use clap::{App, Arg};
+use std::collections::{hash_map::Entry, HashMap};
+use std::io::Cursor;
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+use std::thread::sleep;
+use std::time::Duration;
+
+fn do_main(runnable: Arc<AtomicBool>) -> Result<(), BccError> {
+    let matches = App::new("xdpdropcount")
+        .about("Drop incoming packets on XDP layer and count for which protocol type")
+        .arg(
+            Arg::with_name("device")
+                .long("device")
+                .short("d")
+                .help("Device name to attach the XDP program")
+                .default_value("eth0")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("hardware-offload-mode")
+                .long("hw-offload-mode")
+                .short("h")
+                .help("Run the XDP program in hardware offload mode XDP_FLAGS_HW_MODE")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let hw_offload_mode_enabled = matches.is_present("hw-offload-mode");
```

Comment:
Any reason to have this in a variable as opposed to using `matches.is_present` in the `if` below?

Reply:
The variable is used once more on L48, figured not to repeat the `is_present...` as a result

Reply:
D'oh - you're right. I missed that. Resolving